src/components/app.js

```diff
@@ -192,6 +233,16 @@ function AppPure(props) {
   );
 
   const isDarkMode = window.matchMedia("(prefers-color-scheme: dark)").matches;
+  const filteredList = useMemo(() => {
+    return filter.length > 0
+      ? list.filter((item) =>
+          // TODO: regex?
+          (item.operationName || "")
+            .toLowerCase()
+            .includes(filter.toLowerCase())
```

Comment:
Right, this may well be:

```javascript
(item.operationName || '').match(RegExp(filter, 'i'))
```

:)

Reply:
Uh, actually forgot one thing: since I implemented something similar before, I'd highly suggest to escape the filter by using this litte function (https://stackoverflow.com/a/3561711/2013891) in order to avoid breakage when the user enters special regex char in the filter text, such as `.`, `/`, `[`, etc (I know that could not possibly be in the operation name, but yeah 🙄 ).

```javascript
(item.operationName || '').match(RegExp(escapeRegex(filter), 'i'))

// ...

function escapeRegex(string) {
  return string.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
}
```